dissect/target/plugins/apps/vpns/wireguard.py

```diff
@@ -0,0 +1,150 @@
+"""
+    WireGuard configuration parser.
+
+    Resources:
+    - https://manpages.debian.org/testing/wireguard-tools/wg.8.en.html#CONFIGURATION_FILE_FORMAT
+    - https://github.com/pirate/wireguard-docs
+"""
+
+from collections import OrderedDict
+from typing import Iterator
+from configparser import ConfigParser
+from dissect.target.plugin import Plugin, export, internal
+from dissect.target.helpers.record import TargetRecordDescriptor
+from os.path import basename
+
+WireGuardInterfaceRecord = TargetRecordDescriptor(
+    "application/vpn/wireguard/interface",
+    [
+        ("string", "name"),  # basename of .conf file if unset
+        ("net.ipaddress", "address"),
+        ("string", "private_key"),
+        ("string", "listen_port"),
+        ("string", "fw_mark"),
+        ("string", "dns"),
+        ("varint", "table"),
+        ("varint", "mtu"),
+        ("string", "preup"),
+        ("string", "postup"),
+        ("string", "predown"),
+        ("string", "postdown"),
+        ("string", "source"),
+    ],
+)
+
+WireGuardPeerRecord = TargetRecordDescriptor(
+    "application/vpn/wireguard/peer",
+    [
+        ("string", "name"),
+        ("string", "public_key"),
+        ("string", "pre_shared_key"),
+        ("net.ipnetwork", "allowed_ips"),
+        ("string", "endpoint"),
+        ("varint", "persistent_keep_alive"),
+        ("string", "source"),
+    ],
+)
+
+
+class WireGuardPlugin(Plugin):
+    __namespace__ = "wireguard"
+
+    """
+    # TODO: NetworkManager uses a different stanza format
+    # "/etc/NetworkManager/system-connections/Wireguard*",
+    # TODO: systemd uses a different stanza format
+    # "/etc/systemd/network/wg*.netdev",
+    # "/etc/systemd/network/*wg*.netdev",
+    # TODO: other locations such as $HOME/.config/wireguard
+    # TODO: parse native network manager formats from MacOS, Ubuntu and Windows.
+    """
+    config_globs = [
+        # Linux
+        "/etc/wireguard/*.conf",
+        # Windows
+        "AppData/Local/WireGuard/Configurations/*.conf",
+        # MacOS
+        "/usr/local/etc/wireguard/*.conf",
+        "/opt/homebrew/etc/wireguard/*.conf",
+    ]
+
+    def __init__(self, target) -> None:
+        super().__init__(target)
+        self.configs = []
+
+    def check_compatible(self) -> bool:
+        for path in self.config_globs:
+            cfgs = list(self.target.fs.path().glob(path.lstrip("/")))
+            if len(cfgs) > 0:
+                for cfg in cfgs:
+                    self.configs.append(cfg)
+        if len(self.configs) > 0:
+            return True
+
+    @export(record=WireGuardInterfaceRecord or WireGuardPeerRecord)
+    def config(self) -> Iterator[WireGuardInterfaceRecord or WireGuardPeerRecord]:
```

Comment:
```suggestion
    def config(self) -> Iterator[Union[WireGuardInterfaceRecord, WireGuardPeerRecord]]:
```
Also need to import `Union` from `typing`.